packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -249,6 +250,25 @@ export interface FieldProps {
 	 * @defaultValue If not specified, the key that is persisted is the property key that was specified in the schema.
 	 */
 	readonly key?: string;
+	readonly defaultProvider?: DefaultProvider;
```

Comment:
Updated with documentation

Unfortunately the `NodeKeyManager` and `InsertableContent` are internal, which we need to provide the default values. This was suggested as a workaround to allow `defaultProvider` to be public without updating `NodeKeyManager` and `InsertableContent` to be public

Reply:
That makes sense if we want defaultProvider to be public, but why do we want it to be public? A user declaring their schema with field props will see the `defaultProvider` property, but won't be able to do anything with it, right, because the type is erased? They won't be able to populate the property - it's useless to them. So why do we expose it at all?

Reply:
Updated `SchemaFactory.ts` methods to take in `Omit<FieldProps, "defaultProvider">` for `props` type, to prevent users from manually setting the default provider